### PR TITLE
docs(dx): explain our use of Conventional Commits in CONTRIBUTING.md

### LINKED ISSUE #744: Explain our use of Conventional Commits in `CONTRIBUTING.md`
## Description

This repository uses the Conventional Changelog specification for our commit messages + Pull Request titles. Explain in the `CONTRIBUTING.md` file about how we use these commit messages.

## Implementation model

The Conventional Changelog specification can be read here: https://www.conventionalcommits.org/en/v1.0.0/ - as a refresher:

```
<type>[optional scope]: <description>

[optional body]

[optional footer(s)]
```

The commit types that we use are: `feat`, `fix`, `refactor`, `docs`, `build`, `chore`, `perf`, `style`, `test`. This follows the [Angular commit message guidelines](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type).

If possible, we should also define the **scopes** that this project uses.

## Tasks

- [ ] Explain our use of Conventional Changelog specification
- [ ] Explain the commit types that we use
- [ ] Explain the use of [commitlint](https://commitlint.js.org/#/) to lint commit messages
- [x] (Optional) Switch default merge type for PRs to squash merge-only, to ensure all commit messages match standards

### PR DESCRIPTION
**Pull request by apraditya**
Closes #744 

## Description

Explain our use of Conventional Commits specification

## Current Tasks

- [X] Explain our use of Conventional Changelog specification
- [X] Explain the commit types that we use
- [X] Explain the use of commitlint to lint commit messages
- [X] Translate to Bahasa

### CODE REVIEW

**Comment by resir014** [comment]
@apraditya Don't forget to add the Indonesian version of these guidelines into `CONTRIBUTING_ID.md`. Thanks!

**Comment by apraditya** [comment]
> @apraditya Don't forget to add the Indonesian version of these guidelines into `CONTRIBUTING_ID.md`. Thanks!

Sure @resir014! I'll start the translation once the English is good.

Please check the scopes if I missed something, Also, check if there's one that we shouldn't use or is incorrect format. For example `StackedLink`.

**Comment by resir014** [comment]
@apraditya Yeah, defining all these scopes definitely helps us track down what stuff should we refactor, so thanks for that!

I'll look into this shortly.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -114,6 +119,88 @@ There are several reasons we're using English while communicating in GitHub Issu
...
+- `image`
+- `isoman`
+- `kodiak`
+- `layout`
+- `security`
+- `StackedLink`
+- `seo`
+- `test`
+- `text`
+- `ui`
```
Comment (resir014) [inline CONTRIBUTING.md:202]:
```suggestion
- `ci`
- `assets`
- `components`
  - `donasi`
  - `education`
  - `home`
  - `json-ld`
  - `kontak-darurat`
  - `layout`
  - `telemedicine`
  - `ui`
- `constants`
- `cms`
- `cypress`
- `data`
- `deps`
- `dx`
- `e2e`
- `fetcher`
- `pages`
  - `faq`
  - `isoman`
- `security`
- `seo`
- `ui`
```

@apraditya We can initially start with the top level directories. Bringing the scope levels to a higher level can help reduce the cognitive load on the contributors.

If there's a second level in a list item, only the lowest level gets to be used. But I don't know how to properly convey this.

**Review by resir014 — COMMENTED** [review]
Alright, here's my assessment for now:

File: CONTRIBUTING.md (hunk 1 above)
Comment (apraditya) [reply CONTRIBUTING.md]: > @apraditya I tried narrowing down the scopes a bit. We can initially start with the top level directories. Bringing the scope levels to a higher level can help reduce the cognitive load on the contributors.

Agree

> If there's a second level in a list item, only the lowest level gets to be used. But I don't know how to properly convey this.

Agree with the first sentence. I don't have any preference at the moment. I use the suggestion and put the note about the second level.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -124,6 +129,86 @@ There are several reasons we're using English while communicating in GitHub Issu
...
+To overcome these issues, the maintainers adopt [commit conventions](https://www.conventionalcommits.org/en/)
+to allow contributors to add more semantic meaning to our git history. They use
+[commitlint](https://commitlint.js.org/) to lint the git commit messages.
+To enforce the conventions, the maintainers use a git hook to run the
+commitlint upon a git commit command. The git hook prevents the commit the
+staged files if the commit message failed to comply with the conventions.
+
+It's adviseable to read the [commit conventions](https://www.conventionalcommits.org/en/v1.0.0)
+briefly to further understand the other benefits from them. However, if you're
+having problems with it or what have you, you can use [this
```
Comment (zainfathoni) [inline CONTRIBUTING.md:148]:
```suggestion
having problems with it, you can use [this
```

**Review by zainfathoni — APPROVED** [review]
LGTM 💯 

I just have a minor suggestion.

Thanks for working on it Mas @apraditya @ @resir014!

**Pull request merged**